clients/go/clientutils/clientUtils.go

```diff
@@ -0,0 +1,63 @@
+package clientutils
+
+import (
+	"context"
+	"github.com/flyteorg/flytestdlib/config"
+	grpcMiddleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpcRetry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	grpcPrometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type Config struct {
+	Endpoint              config.URL      `json:"endpoint" pflag:",For admin types, specify where the uri of the service is located."`
+	UseInsecureConnection bool            `json:"insecure" pflag:",Use insecure connection."`
+	MaxBackoffDelay       config.Duration `json:"maxBackoffDelay" pflag:",Max delay for grpc backoff"`
+	PerRetryTimeout       config.Duration `json:"perRetryTimeout" pflag:",gRPC per retry timeout"`
+	MaxRetries            int             `json:"maxRetries" pflag:",Max number of gRPC retries"`
+}
```

Comment:
Having a config in util file seems odd. Also file name are followed to have snakeCase